src/test/java/com/iota/iri/service/tipselection/impl/ConnectedComponentsCalculatorTest.java

```diff
@@ -0,0 +1,240 @@
+
+        ConnectedComponentsCalculator connectedComponentsCalculator = new ConnectedComponentsCalculator(tangle);
+        Collection<Hash> recentTransactions = connectedComponentsCalculator.findNMostRecentTransactions(
+                Collections.singleton(tipsViewModel.getRandomSolidTipHash()));
+        Assert.assertFalse(recentTransactions.contains(Hash.NULL_HASH));
+    }
+
+    @Test
+    public void starAroundGenesis() throws Exception {
+        long incr = 1000;
+
+        final int amount = ConnectedComponentsCalculator.N - 1;
+        List<TransactionViewModel> transactions = new ArrayList<TransactionViewModel>();
+
+        for (int i = 0; i < amount; i++) {
+            transactions.add(new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(
+                    Hash.NULL_HASH,
+                    Hash.NULL_HASH),
+                    getRandomTransactionHash()));
+        }
+
+        for (TransactionViewModel transaction : transactions) {
+            transaction.setArrivalTime(incr++);
+            transaction.store(tangle);
+        }
+
+        Collection<Hash> transactionHashes = transactions.stream().map(TransactionViewModel::getHash).collect(Collectors.toList());
+
+        ConnectedComponentsCalculator connectedComponentsCalculator = new ConnectedComponentsCalculator(tangle);
+        Collection<Hash> recentTransactions = connectedComponentsCalculator.findNMostRecentTransactions(
+                transactionHashes);
+        Assert.assertTrue(recentTransactions.containsAll(transactionHashes));
+        Assert.assertTrue(recentTransactions.contains(Hash.NULL_HASH));
+
+    }
+
+    @Test
+    public void starAroundGenesisTooBig() throws Exception {
+        long incr = 1000;
+
+        final int amount = ConnectedComponentsCalculator.N + 1;
+        List<TransactionViewModel> transactions = new ArrayList<TransactionViewModel>();
+
+        for (int i = 0; i < amount; i++) {
+            transactions.add(new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(
+                    Hash.NULL_HASH,
+                    Hash.NULL_HASH),
+                    getRandomTransactionHash()));
+        }
+
+        for (TransactionViewModel transaction : transactions) {
+            transaction.setArrivalTime(incr++);
+            transaction.store(tangle);
+        }
+
+        Collection<Hash> transactionHashes = transactions.stream().map(TransactionViewModel::getHash).collect(Collectors.toList());
+
+        ConnectedComponentsCalculator connectedComponentsCalculator = new ConnectedComponentsCalculator(tangle);
+        Collection<Hash> recentTransactions = connectedComponentsCalculator.findNMostRecentTransactions(
+                transactionHashes);
+        Assert.assertFalse(recentTransactions.contains(Hash.NULL_HASH));
+        Assert.assertFalse(recentTransactions.containsAll(transactionHashes)); // there exists one tip not in list
```

Comment:
Instead of this assertion (or maybe in addition), I'd put the more specific assertion that the oldest tx is the one that's excluded.